packages/charts/src/vaadin-chart.js

```diff
@@ -1348,6 +1348,7 @@ class Chart extends ResizeMixin(ElementMixin(ThemableMixin(PolymerElement))) {
               this.tempBodyStyle = document.createElement('style');
               this.tempBodyStyle.textContent = effectiveCss;
               document.body.appendChild(this.tempBodyStyle);
+              document.body.setAttribute('styled-mode', this.options.chart.styledMode.toString());
```

Comment:
This would set `styled-mode="false"` in case when styled mode is not enabled:
![Screenshot 2022-08-03 at 10 28 22](https://user-images.githubusercontent.com/10589913/182549654-a4f7cbcb-3e7b-4a4b-9d40-edf19ac909ea.png)

This is wrong, because this attribute would still make `:where([styled-mode])` CSS apply.

Let's instead check if `styledMode` is set before setting attribute:

```js
if (this.options.chart.styledMode) {
  document.body.setAttribute('styled-mode', '');
}
```

And then also update the corresponding block below to check if the attribute should be removed.

Reply:
Addressed with the latest commits